zebra-state/src/service/queued_blocks.rs

```diff
@@ -219,18 +219,49 @@ impl QueuedBlocks {
     }
 }
 
+#[derive(Debug)]
+struct HashAndHeight {
+    hash: block::Hash,
+    height: block::Height,
+}
+
+impl From<(block::Hash, block::Height)> for HashAndHeight {
+    fn from((hash, height): (block::Hash, block::Height)) -> Self {
+        Self { hash, height }
+    }
+}
+
+#[derive(Debug)]
+enum SentHash {
+    Finalized(HashAndHeight),
+    NonFinalized(HashAndHeight),
+}
+
+impl std::ops::Deref for SentHash {
+    type Target = HashAndHeight;
+
+    fn deref(&self) -> &Self::Target {
+        let (Self::Finalized(v) | Self::NonFinalized(v)) = self;
+        v
+    }
+}
+
 #[derive(Debug, Default)]
 pub(crate) struct SentHashes {
     /// A list of previously sent block batches, each batch is in increasing height order.
     /// We use this list to efficiently prune outdated hashes that are at or below the finalized tip.
-    bufs: Vec<VecDeque<(block::Hash, block::Height)>>,
+    bufs: Vec<VecDeque<SentHash>>,
 
     /// The list of blocks sent in the current batch, in increasing height order.
-    curr_buf: VecDeque<(block::Hash, block::Height)>,
+    curr_buf: VecDeque<SentHash>,
+
+    /// Stores a set of hashes that may have been sent to the block write task via
+    /// the finalized block write channel but may not be in the finalized state yet.
+    sent_finalized: HashMap<block::Hash, Vec<transparent::OutPoint>>,
 
-    /// Stores a set of hashes that have been sent to the block write task but
-    /// may not be in the finalized state yet.
-    sent: HashMap<block::Hash, Vec<transparent::OutPoint>>,
+    /// Stores a set of hashes that have been sent to the block write task via the
+    /// non-finalized block write channel but may not be in the finalized state yet.
```

Comment:
Which state do you mean here?

Reply:
The finalized state, I'll add some context on how it's used in `can_fork_chain_at()`.

Reply:
```suggestion
    /// non-finalized block write channel, but haven't been committed to the finalized state yet.
    /// (They are either in the channel, or in the non-finalized state.)
```

Reply:
I wasn't asking for a code change here, I just wanted to check that "finalized state" wasn't a copy-paste error.

Reply:
> I wasn't asking for a code change here

I know - but I think I initially overcomplicated this, I updated this so that it prunes out all the finalized sent hashes and marks a `can_fork_chain_at_hashes` flag as true. It should actually just drop it and replace it with a new one since it'll always be emptied anyways. I'll do this in a follow-up PR.